app/aws/s3.py

```diff
@@ -108,6 +135,20 @@ def get_phone_number_from_s3(service_id, job_id, job_row_number):
     else:
         incr_jobs_cache_hits()
 
+    if job is None:
+        current_app.logger.warning(
+            "Couldnt find phone for job_id {job_id} row number {job_row_number} because job is missing"
+        )
+        return "Unknown Phone"
+
+    if JOBS.get(f"{job_id}_phones") is None:
+        JOBS[f"{job_id}_phones"] = extract_phones(job)
+
+    if JOBS.get(f"{job_id}_phones") is not None:
+        phone_to_return = JOBS.get(f"{job_id}_phones").get(job_row_number)
+        if phone_to_return:
+            return phone_to_return
+
```

Comment:
Just making sure I understand this flow correctly:

- Check if the job exists first and if not, just immediately return with unknown phone because we can't get it
- If there is a job, try and find the phone numbers in the `ExpiringDict`
   - If it's not found, try and extract all the phone numbers to populate the dictionary
- Try to find the phone numbers again in case they had to be extracted previously
   - If there is a phone match, return it here
   - If not, continue processing as before, and if the phone still can't be found, return unknown phone again

I'm double checking to make sure I understand because I'm wondering if we need the processing we were doing before, or if the `ExpiringDict` would cover whatever we need if it can be populated from the job; and if it can't be populated, then there's nothing else to do, right?